### DISCUSSION TITLE
ruby ile ilgili snake_case ve missing top-level documantation arızası

### DISCUSSION THREAD

**Discussion by hatqi** [19/classroom#34] [Anonim] [answered]
![image](https://user-images.githubusercontent.com/100834264/158687661-adbb2290-5160-41a6-b4fe-091c47dcd284.png)
![image](https://user-images.githubusercontent.com/100834264/158685950-e904a6e7-f12d-4f6a-8df5-9bb988be3883.png)
buradaki sorunu bir türlü çözemedim. bir çıkış yolu bulamadım. sorunun programın çalışmasında engel teşkil etmiyor ancak ödev testinde 40/100 çıkıyor. çözüm önerileriniz için şimdiden teşekkürler.

**Comment by roktas** [comment]
Foruma çözüm göndermeyin. Sorununa bakacağım.

**Comment by roktas** [comment, marked as answer]
- Rubocop'la ilgili sorunun ödev değerlendirmesinde bir önemi yok, göz ardı edebilirsin.
- Çözümün hatalı, düzeltmen gereken ciddi problemler var.  Öncelikle ona yoğunlaş.  İpucu: (1) hatalı yerlerde `puts` yapıyorsun, (2) metodların dönüş değerine (tipine) dikkat et

**Reply by hatqi** [reply]
cevabınız ve yönlendirmeniz için teşekkür ederim. Sorunu mümkün olduğunca az puts komutunu kullanarak çözdüm. 
Ancak konu ile ilgili bir değerlendirme yapmak gerekirse;
Karşılaşılan Zorluklar;
-> Derlemede hata vermeden çalışıp otomatik değerlendirmede kalması,
-> Yazım kurallarının compile aşamasında sorun çıkartmamasına rağmen otomatik değerlendirmede dikkate alınması,
-> Bazı kuralların çok katı olması (örneğin boşluklar ve bu boşlukların yine compile aşamasında sorun çıkarmaması) 
-> Compile aşamasında sorun olmadan çalışan programın değerlendirme de kalması sebebiyle çok fazla zaman alması
 açıkçası ruby'e karşı ilk izlenimimi olumsuz etkilediğini ifade etmek isterim. 
Faydaarı;
-> Yazım kurallarının dil bilgisi kurallarına yakın olması sebebiyle kod yazmak biraz daha kolay,

 iyi çalışmalar dilerim.

**Reply by roktas** [reply]
@hatqi "Compile" vs diye bir aşama yok, değerlendirme yapan bir dizi test var sadece.  Bu testler birbirinden bağımsız, birinden başarılı olman diğerinden başarılı olacağın anlamına gelmiyor.

Dil ve/veya kurallar hakkında bu tür "büyük" yargılar geliştirmen için yeterli bilgi ve deneyime sahip değilsin.  Bunu böyle kafanda sorgulamak yerine daha fazla çalışmalı, daha çok şey öğrenmelisin.  Yaşadığın sorun "çok puts" kullanmak değildi, düpedüz hatalı kod yazmıştın ve bunun dille/kurallarla ilgisi yok. Bir başka şekilde söylemem gerekirse "seveceğin" bir başka dil veya kural setiyle de aynı hatayı yapardın. :-)  Umarım bunu anlamışsındır (veya bu açıklamalardan sonra biraz daha düşünüp anlarsın).

Dipnot: Mesajında listeleme yapmaya çalışmışsın.  Farkettiysen metin alanlarında "Preview" diye bir sekme var ve bu "bir nedenle" var.  [Markdown sözdizimi kurallarını](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) öğrenirsen yazdığın metin daha düzgün görünür.  Örnek liste (karşılaştır):

- Derlemede hata vermeden çalışıp otomatik değerlendirmede kalması,
- Vesaire
